eva_submission/eload_submission.py

```diff
@@ -57,6 +54,72 @@ def _get_dir(self, key):
     def now(self):
         return datetime.now()
 
+    def upgrade_config_if_needed(self, analysis_alias=None):
+        """
+        Upgrades unversioned configs (i.e. pre-1.0) to the current version, making a backup first and using the
+        provided analysis alias for all vcf files. Currently doesn't perform any other version upgrades.
+        """
+        if 'version' not in self.eload_cfg:
+            self.info(f'No version found in config, upgrading to version {__version__}.')
+            self.eload_cfg.backup()
+
+            if 'submission' not in self.eload_cfg:
+                self.error('Need submission config section to upgrade')
+                self.error('Try running prepare_submission or prepare_backlog_study to build a config from scratch.')
+                raise ValueError('Need submission config section to upgrade')
+
+            # Note: if we're converting an old config, there's only one analysis
+            if not analysis_alias:
+                analysis_alias = self._get_analysis_alias_from_metadata()
+            analysis_data = {
+                'assembly_accession': self.eload_cfg.pop('submission', 'assembly_accession'),
+                'assembly_fasta': self.eload_cfg.pop('submission', 'assembly_fasta'),
+                'assembly_report': self.eload_cfg.pop('submission', 'assembly_report'),
+                'vcf_files': self.eload_cfg.pop('submission', 'vcf_files')
+            }
+            analysis_dict = {analysis_alias: analysis_data}
+            self.eload_cfg.set('submission', 'analyses', value=analysis_dict)
+
+            if 'validation' in self.eload_cfg:
+                self.eload_cfg.pop('validation', 'valid', 'vcf_files')
+                self.eload_cfg.set('validation', 'valid', 'analyses', value=analysis_dict)
+
+            if 'brokering' in self.eload_cfg:
+                brokering_vcfs = {
+                    vcf_file: index_dict
+                    for vcf_file, index_dict in self.eload_cfg.pop('brokering', 'vcf_files').items()
+                }
+                brokering_analyses = deepcopy(analysis_dict)
+                brokering_analyses[analysis_alias]['vcf_files'] = brokering_vcfs
+                self.eload_cfg.set('brokering', 'analyses', value=brokering_analyses)
+                analysis_accession = self.eload_cfg.pop('brokering', 'ena', 'ANALYSIS')
+                self.eload_cfg.set('brokering', 'ena', 'ANALYSIS', analysis_alias, value=analysis_accession)
+
+            # Set version once we've successfully upgraded
+            self.eload_cfg.set('version', value=__version__)
+        else:
+            self.info(f"Config is version {self.eload_cfg.query('version')}, not upgrading.")
+
+    def _get_analysis_alias_from_metadata(self):
```

Comment:
I know it's a bit more involved but it is safer to get the analysis alias from ENA than from the spreadsheet.

Reply:
I think the analysis alias is only useful during the pre-brokering steps, after that we should be using the analysis accession from ENA rather than the alias.

I haven't thought about what would happen if the analysis alias is changed in ENA directly after it's already received an accession but before ingestion is run... I think the ingestion would break in that case, but hopefully that doesn't happen too often.

Reply:
Yes you're right.